Could not build anrdoid-app

```
Starting a Gradle Daemon (subsequent builds will be faster)
> Task :wrapper
BUILD SUCCESSFUL in 14s
1 actionable task: 1 executed
Errors during XML parse:
Additionally, the fallback loader failed to parse the XML.
Checking the license for package Android SDK Platform 30 in /usr/local/lib/android/sdk/licenses
License for package Android SDK Platform 30 accepted.
Preparing "Install Android SDK Platform 30 (revision 3)".
"Install Android SDK Platform 30 (revision 3)" ready.
Installing Android SDK Platform 30 in /usr/local/lib/android/sdk/platforms/android-30
"Install Android SDK Platform 30 (revision 3)" complete.
"Install Android SDK Platform 30 (revision 3)" finished.
Checking the license for package Android SDK Platform-Tools in /usr/local/lib/android/sdk/licenses
License for package Android SDK Platform-Tools accepted.
Preparing "Install Android SDK Platform-Tools v.35.0.2".
"Install Android SDK Platform-Tools v.35.0.2" ready.
Installing Android SDK Platform-Tools in /usr/local/lib/android/sdk/platform-tools
"Install Android SDK Platform-Tools v.35.0.2" complete.
"Install Android SDK Platform-Tools v.35.0.2" finished.
> Task :app:preBuild UP-TO-DATE
> Task :app:preDebugBuild UP-TO-DATE
> Task :app:mergeDebugNativeDebugMetadata NO-SOURCE
> Task :app:checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :app:dataBindingMergeDependencyArtifactsDebug
> Task :app:generateDebugResValues
> Task :app:generateDebugResources
> Task :app:generateDebugBuildConfig
> Task :app:mapDebugSourceSetPaths
> Task :app:createDebugCompatibleScreenManifests
> Task :app:extractDeepLinksDebug
> Task :app:checkDebugAarMetadata FAILED
> Task :app:processDebugMainManifest FAILED
FAILURE: Build completed with 2 failures.
1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:checkDebugAarMetadata'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
   > 20 issues were found when checking AAR metadata:
> Task :app:mergeDebugResources
     
       1.  Dependency 'androidx.appcompat:appcompat-resources:1.7.0' requires libraries and applications that
           depend on it to compile against version 34 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 34, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
       2.  Dependency 'androidx.appcompat:appcompat:1.7.0' requires libraries and applications that
           depend on it to compile against version 34 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 34, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
       3.  Dependency 'androidx.emoji2:emoji2-views-helper:1.3.0' requires libraries and applications that
           depend on it to compile against version 33 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 33, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
       4.  Dependency 'androidx.emoji2:emoji2:1.3.0' requires libraries and applications that
           depend on it to compile against version 33 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 33, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
       5.  Dependency 'androidx.fragment:fragment:1.5.4' requires libraries and applications that
           depend on it to compile against version 31 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
10 actionable tasks: 10 executed
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 31, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
       6.  Dependency 'androidx.activity:activity:1.8.0' requires libraries and applications that
           depend on it to compile against version 34 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 34, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
       7.  Dependency 'androidx.transition:transition:1.5.0' requires libraries and applications that
           depend on it to compile against version 34 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 34, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
       8.  Dependency 'androidx.core:core:1.13.1' requires libraries and applications that
           depend on it to compile against version 34 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 34, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
       9.  Dependency 'androidx.savedstate:savedstate:1.2.1' requires libraries and applications that
           depend on it to compile against version 33 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 33, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
      10.  Dependency 'androidx.lifecycle:lifecycle-livedata:2.6.2' requires libraries and applications that
           depend on it to compile against version 33 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 33, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
      11.  Dependency 'androidx.lifecycle:lifecycle-viewmodel:2.6.2' requires libraries and applications that
           depend on it to compile against version 33 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 33, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
      12.  Dependency 'androidx.lifecycle:lifecycle-process:2.6.2' requires libraries and applications that
           depend on it to compile against version 33 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 33, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
      13.  Dependency 'androidx.lifecycle:lifecycle-livedata-core:2.6.2' requires libraries and applications that
           depend on it to compile against version 33 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 33, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
      14.  Dependency 'androidx.lifecycle:lifecycle-runtime:2.6.2' requires libraries and applications that
           depend on it to compile against version 33 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 33, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
      15.  Dependency 'androidx.lifecycle:lifecycle-viewmodel-savedstate:2.6.2' requires libraries and applications that
           depend on it to compile against version 33 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 33, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
      16.  Dependency 'androidx.core:core-ktx:1.13.1' requires libraries and applications that
           depend on it to compile against version 34 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 34, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
      17.  Dependency 'androidx.annotation:annotation-experimental:1.4.0' requires libraries and applications that
           depend on it to compile against version 34 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 34, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
      18.  Dependency 'androidx.profileinstaller:profileinstaller:1.3.1' requires libraries and applications that
           depend on it to compile against version 33 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 33, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
      19.  Dependency 'androidx.startup:startup-runtime:1.1.1' requires libraries and applications that
           depend on it to compile against version 31 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 31, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
     
      20.  Dependency 'androidx.arch.core:core-runtime:2.2.0' requires libraries and applications that
           depend on it to compile against version 33 or later of the
           Android APIs.
     
           :app is currently compiled against android-30.
     
           Recommended action: Update this project to use a newer compileSdk
           of at least 33, for example 35.
     
           Note that updating a library or application's compileSdk (which
           allows newer APIs to be used) can be done separately from updating
           targetSdk (which opts the app in to new runtime behavior) and
           minSdk (which determines which devices the app can be installed
           on).
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org./
==============================================================================
2: Task failed with an exception.
-----------
* What went wrong:
A problem was found with the configuration of task ':app:processDebugMainManifest' (type 'ProcessApplicationManifest').
  - In plugin 'com.android.internal.version-check' type 'com.android.build.gradle.tasks.ProcessApplicationManifest' property 'mainManifest' specifies file '/home/runner/work/location-app/location-app/frontend/android-app/app/src/main/AndroidManifest.xml' which doesn't exist.
    
    Reason: An input file was expected to be present but it doesn't exist.
    
    Possible solutions:
      1. Make sure the file exists before the task is called.
      2. Make sure that the task which produces the file is declared as an input.
    
    For more information, please refer to https://docs.gradle.org/8.10.2/userguide/validation_problems.html#input_file_does_not_exist in the Gradle documentation.
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org./
==============================================================================
BUILD FAILED in 14s
Error: Process completed with exit code 1.
```

--- a/frontend/android-app/app/build.gradle.kts
+++ b/frontend/android-app/app/build.gradle.kts
@@ -49,7 +49,6 @@ android {
         abortOnError = false
     }
     kotlinOptions {
-        jvmTarget = "22"
-        freeCompilerArgs += listOf("-Xjvm-default=compatibility")
+        jvmTarget = "21"
     }
 }

--- a/frontend/android-app/app/build.gradle.kts
+++ b/frontend/android-app/app/build.gradle.kts
@@ -16,11 +16,11 @@ dependencies {
 
 android {
     namespace = "com.example.locationapp"
-    compileSdk = 30
+    compileSdk = 35
     defaultConfig {
         applicationId = "com.example.locationapp"
         minSdk = 21
-        targetSdk = 30
+        targetSdk = 35
         versionCode = 1
         versionName = "1.0"
         buildConfigField("String", "API_BASE_URL", "\"https://api.example.com/\"")

--- a/frontend/android-app/app/build.gradle.kts
+++ b/frontend/android-app/app/build.gradle.kts
@@ -14,6 +14,10 @@ dependencies {
     androidTestImplementation("androidx.test.espresso:espresso-core:3.6.1")
 }
 
+kotlin {
+    jvmToolchain(21)
+}
+
 android {
     namespace = "com.example.locationapp"
     compileSdk = 35
@@ -47,8 +51,5 @@ android {
     }
     lint {
         abortOnError = false
-    }
-    kotlinOptions {
-        jvmTarget = "21"
     }
 }

--- a/frontend/android-app/app/src/main/AndroidManifest.xml
+++ b/frontend/android-app/app/src/main/AndroidManifest.xml
@@ -9,7 +9,7 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name=".MainActivity">
+        <activity android:name=".MainActivity" android:exported="true">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
                 <category android:name="android.intent.category.LAUNCHER" />

--- a/frontend/android-app/app/src/main/AndroidManifest.xml
+++ b/frontend/android-app/app/src/main/AndroidManifest.xml
@@ -1,6 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.example.locationapp">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
 
     <application>
         <activity android:name=".MainActivity" android:exported="true">

--- a/frontend/android-app/app/src/main/AndroidManifest.xml
+++ b/frontend/android-app/app/src/main/AndroidManifest.xml
@@ -2,13 +2,7 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.example.locationapp">
 
-    <application
-        android:allowBackup="true"
-        android:icon="@mipmap/ic_launcher"
-        android:label="@string/app_name"
-        android:roundIcon="@mipmap/ic_launcher_round"
-        android:supportsRtl="true"
-        android:theme="@style/AppTheme">
+    <application>
         <activity android:name=".MainActivity" android:exported="true">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />

--- a/frontend/android-app/app/src/main/AndroidManifest.xml
+++ b/frontend/android-app/app/src/main/AndroidManifest.xml
@@ -1,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.example.locationapp">
+
+    <application
+        android:allowBackup="true"
+        android:icon="@mipmap/ic_launcher"
+        android:label="@string/app_name"
+        android:roundIcon="@mipmap/ic_launcher_round"
+        android:supportsRtl="true"
+        android:theme="@style/AppTheme">
+        <activity android:name=".MainActivity">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
+    </application>
+
+</manifest>

